### ISSUE TITLE
error 'Did you start 'gksqt'?  after Pkg.update()  to GR ver 0.31

### ISSUE THREAD

**Issue by programistawpf** [jheinen/GR.jl#111]
after Pkg.update()  to GR ver 0.31 
I have error :

`julia> using Plots ; gr()
Plots.GRBackend()

julia> plot()
connect: No error
GKS: can't connect to GKS socket application
Did you start 'gksqt'?

GKS: Open failed in routine OPEN_WS
GKS: GKS not in proper state. GKS must be either in
GKS: GKS not in proper state. GKS must be either in
GKS: GKS not in proper state. GKS must be either in
GKS: GKS not in proper state. GKS must be either in`
Paul

**Comment by jheinen** [comment]
What happens if you manually start `gksqt`? I suspect the Qt5 run-time is missing on you system ... Are you on a Linux box?

**Comment by programistawpf** [comment]
Win7 64. The error appears randomly, but quite often.

**Comment by programistawpf** [comment]
_
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.2 (2017-12-13 18:08 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

**Comment by programistawpf** [comment]
And interesting is that not displayed charts is posible to save by "savefig" to pdf ;)

**Comment by jheinen** [comment]
That makes sense, as GR has its own drivers for PDF, etc.
Is the error time dependent? Does it only occur if you use it (let's say) several times a minute? If so, the reason might be a problem in the socket communication - sockets might be blocked (for about 30 seconds). Could you check this. I only have a Windows 10 installation and need some hint to get into the right direction ...

**Comment by programistawpf** [comment]
Thanks.
" socket communication " in WIn or in Julia? How to checjk it ?

**Comment by jheinen** [comment]
The question was: Do you have the impression that the problem **only** occurs when you repeatedly use GR (let's say 2-3 times a minute or in intervals < 30s)

**Comment by programistawpf** [comment]
Not . Some times error is in first using :  plot() .  If at first using plot() is ok then I can create new empty plot very often , 1/sek, many times likt this:
julia> using Plots ; gr()
Plots.GRBackend()
julia> plot()
julia> plot()
julia> plot()
julia> plot()
julia> plot()

and empty plot still exist

**Comment by programistawpf** [comment]
using Plots ; gr()
for i=1:1000
display(plot())
sleep(5)
end

It looks more like a start in a bad moment. In the loop, the diagram is sometimes different, eg without scale  or all gray :/

**Comment by programistawpf** [comment]
for i=1:1000
using Plots ; gr();display(plot())
sleep(0.5)
end

In this loop anomalias are ONLY in +- 9 firs displays AFTER closing plot manualy (and i firts 9 displays on the bigining of loop. If plot is not closed , and  the loop is broken manulay in the next  playing  loop no anomalias .

**Comment by programistawpf** [comment]
After this error, when startin loop with :
p1=plot(map(Date,czastemp),dane,label=string("dane"),title=etykiety[j],linecolor = :blue,ylims = (minimum(dane),maximum(dane)));

p2=plot(map(Date,czas)[end-11:end],dane[end-11:end],linecolor = :blue,legend = false,xticks=map(Date,czas)[end-prognoza.*2+1:end][1:12:end],ylims = (minimum(dane),maximum(dane)));

display(plot(p1,p2,layout=(2,1),wsize=(600,600)))

is error 
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYMARKER

**Comment by jheinen** [comment]
I have to debug this when I'm back in office. I  currently only have access to a Windows 10 VM.

**Comment by programistawpf** [comment]
Sometime !!! error back :/ 

julia> versioninfo()
Julia Version 1.0.4
Commit 38e9fb7f80 (2019-05-16 03:38 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-2630QM CPU @ 2.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, sandybridge)

Win7 64


for i in 1:1555
display(plot(rand(5)))
end
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE

I checed many new sesions of Julia , Error i s RANDOM ! 
(v1.0) pkg> st
    Status `C:\Users\PC\.julia\environments\v1.0\Project.toml`
  [f9309374] ApplicationBuilder v0.4.0
  [7d9fca2a] Arpack v0.3.1
  [6e4b80f9] BenchmarkTools v0.4.2
  [a93c6f00] DataFrames v0.19.2
  [28b8d3ca] GR v0.41.0
  [f67ccb44] HDF5 v0.12.1
  [7073ff75] IJulia v1.19.0
  [a15396b6] OnlineStats v0.27.0
  [91a5bcdd] Plots v0.26.2
  [d71aba96] ReadStat v0.4.1
  [1463e38c] StatFiles v0.8.0
  [2913bbd2] StatsBase v0.32.0
  [c17dfb99] WinRPM v0.4.2
  [fdbf4ff8] XLSX v0.5.4

Please help, sometime i nmust works on Win 7 .
Paul

**Comment by xiaodaigh** [comment]
> What happens if you manually start `gksqt`?

How do you manually start `gksqt`?

**Comment by programistawpf** [comment]
which to use ? the youngest ? 
![GR](https://user-images.githubusercontent.com/5564741/71680261-ea20a980-2d89-11ea-8c1b-4c19b0ca8fef.png)

**Comment by jheinen** [comment]
Yes. The other directories can be removed. No idea, why they still exist!?

**Comment by programistawpf** [comment]
On my win 7 I have many verson Julia 1.0.4 1  ; 1.0.5  ; 1.2.... more... 
   Maybe for this ?

TEST:
After run gksqt.exe , but not youngest, second of them sorted by time 
(DOS window opened and close) plots looks OK :) 7 for 7 times works ok! 
Big Thanks!

This path help me:
C:\Users\PC\.julia\packages\GR\ZI5OE\deps\gr\bin

How to reconize links before version of Julai with version of gksqt.exe 
on my machine ?
Paul

W dniu 2020-01-02 o 18:40, Josef Heinen pisze:

**Comment by jheinen** [comment]
On Windows the `gksqt.exe` binary is located in the same directory as the GR run-time (`libgr.dll`) you are using. In your case, as you seem to have multiple Julia versions, there should be only one(!) GR folder for each Julia distribution.
